docs/developer-guide/base-maps/using-with-maplibre.md

```diff
@@ -0,0 +1,248 @@
+
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs groupId="language">
+  <TabItem value="ts" label="TypeScript">
+
+```ts
+import {MapboxOverlay} from '@deck.gl/mapbox';
+import {ScatterplotLayer} from '@deck.gl/layers';
+import maplibregl from 'maplibre-gl';
+import 'maplibre-gl/dist/maplibre-gl.css';
+
+const map = new maplibregl.Map({
+  container: 'map',
+  style: 'https://basemaps.cartocdn.com/gl/positron-gl-style/style.json',
+  center: [0.45, 51.47],
+  zoom: 11
+});
+
+map.once('load', () => {
+  const deckOverlay = new MapboxOverlay({
+    interleaved: true,
+    layers: [
+      new ScatterplotLayer({
+        id: 'deckgl-circle',
+        data: [
+          {position: [0.45, 51.47]}
+        ],
+        getPosition: d => d.position,
+        getFillColor: [255, 0, 0, 100],
+        getRadius: 1000,
+        beforeId: 'watername_ocean' // In interleaved mode render the layer under map labels
+      })
+    ]
+  });
+
+  map.addControl(deckOverlay);
+});
+```
+
+  </TabItem>
+  <TabItem value="react" label="React">
+
+```tsx
+import React from 'react';
+import {Map, useControl} from 'react-map-gl/maplibre';
+import {MapboxOverlay} from '@deck.gl/mapbox';
+import {DeckProps} from '@deck.gl/core';
+import {ScatterplotLayer} from '@deck.gl/layers';
+import 'maplibre-gl/dist/maplibre-gl.css';
+
+function DeckGLOverlay(props: DeckProps) {
+  const overlay = useControl<MapboxOverlay>(() => new MapboxOverlay(props));
+  overlay.setProps(props);
+  return null;
+}
+
+function App() {
+  const layers: [
+    new ScatterplotLayer({
+      id: 'deckgl-circle',
+      data: [
+        {position: [0.45, 51.47]}
+      ],
+      getPosition: d => d.position,
+      getFillColor: [255, 0, 0, 100],
+      getRadius: 1000,
+      beforeId: 'watername_ocean' // In interleaved mode render the layer under map labels
+    })
+  ];
+
+  return (
+    <Map
+      initialViewState={{
+        longitude: 0.45,
+        latitude: 51.47,
+        zoom: 11
+      }}
+      mapStyle="https://basemaps.cartocdn.com/gl/positron-gl-style/style.json"
+    >
+      <DeckGLOverlay layers={layers} interleaved />
+    </Map>
+  );
+}
+```
+
+  </TabItem>
+</Tabs>
+
+
+You can find full project setups in the [react get-started example](https://github.com/visgl/deck.gl/tree/master/examples/get-started/react/maplibre/) and [pure js get-started example](https://github.com/visgl/deck.gl/tree/master/examples/get-started/pure-js/maplibre/).
+
+
+### Example: reverse controlled
+
+The reverse-controlled option is supported by the pre-built scripting bundle, and in React when used with the `react-map-gl` library. There is currently no easy way to do it under Vanilla JS.
+
+<Tabs groupId="language">
+  <TabItem value="scripting" label="Scripting">
+
+```js
+<script src="https://unpkg.com/deck.gl@^9.0.0/dist.min.js"></script>
+<script src="https://unpkg.com/maplibre-gl@^3.0.0/dist/maplibre-gl.js"></script>
```

Comment:
Consider linking to version 4, I don't know if this was tested or not, but I think it's better to link to the latest version if possible.